frontend/src/app/components/network-graph/network-graph.component.ts

```diff
@@ -213,9 +213,14 @@ export class NetworkGraphComponent implements OnInit, OnChanges, OnDestroy, Afte
     this.subscription = this.route.queryParamMap
       .pipe(
         map(params => params.get('node')),
-        map(nodeAddress =>
-          this._filter(nodeAddress).filter(element => element.type === ElementType.NODE)
-        ),
+        filter(value => !!value),
+        map(query => {
+          const results = this._filter(query).filter(element => element.type === ElementType.NODE);
+          if (results.length !== 1) {
+            this.showQueryError(query);
+          }
+          return results;
+        }),
         filter(results => results.length === 1),
```

Comment:
I guess this line can be now removed since the `map` is already doing this?

Reply:
No, because the `map` still returns invalid results. We still need to filter them. Maybe it's a bit ugly to have two checks.